### ISSUE TITLE
Adding Heavy-hex lattice

### ISSUE THREAD

**Issue by PhilipVinc** [netket/netket#1530]
I would like to add the heavy hex lattice as a built-in one in netket to make it easy to look at ibm's stuff. 

This is relatively easy to do, thanks to @attila-i-szabo  implementation of lattice, and I think can be done by adding two new sites to the unit cell 

```python
import netket as nk

extent = (2,2)
site_offsets = [
    [0.50, 0.50 / 3**0.5],
    [0.75, 0.75 / 3**0.5],
    [1.00, 1.00 / 3**0.5],
    [1.25, 0.75 / 3**0.5],
    [1.00, 1.50 / 3**0.5],
]

g = nk.graph._common_lattices._hexagonal_general(
    extent=extent,
    site_offsets=site_offsets,
    pbc=False,
)
```

but this gives the following graph

<img width="752" alt="Screenshot 2023-07-13 at 13 44 46" src="https://github.com/netket/netket/assets/2407108/d13fd473-5977-496d-ac13-3e3b58d60c20">

which is correct, but does not really match the way that IBM uses it (also, the 'unit' used by ibm corresponds to 2 unit cells in this representation.

But the way IBM uses this lattice is that... they have a certain number of unit cells plus some extra sites. 
I'm unsure how to do it exactly if we want to keep using a Lattice...
Indeed we could just hardcode the same graph structure that they use, but maybe I'm missing something...

**Comment by PhilipVinc** [comment]
almost. The shape used by ibm devices is this one, which as you correctly point out is not really a periodic lattice because of the extra sites

![41586_2023_6096_Fig1_HTML png copy](https://github.com/netket/netket/assets/2407108/d9c42136-09bc-4fb0-8ec9-1c83db54e5fd)

**Comment by attila-i-szabo** [comment]
Yeah. Idk what the best way of constructing this would be - quite possibly just a snaking chain + vertical bonds added by hand

**Comment by PhilipVinc** [comment]
[retworkx](https://qiskit.org/documentation/retworkx/dev/apiref/rustworkx.generators.heavy_hex_graph.html) , IBM's reimplementation of `networkx` in rust, does generate their interpretation of an heavy-hex graph, but the code is messy. Someone should either ask gpt4 or take a student and make him write it.

**Comment by gcarleo** [comment]
if that code generates an adjacency list... I think we don't need to implement it in netket, I would just call that code and put the adjacency list in one of our graph-based objects ?

**Comment by PhilipVinc** [comment]
and erroring if the user has not installed retworkx? indeed that's the simplest option.
